app/controllers/user/courses_controller.rb

```diff
@@ -1,8 +1,8 @@
 class User::CoursesController < User::ApplicationController
   skip_before_action :authenticate_user!, only: %i(index)
-  skip_before_action :ensure_user_role, only: %i(index)
   before_action :redirect_guest_status_param, only: %i(index)
-  before_action :set_course, only: %i(show enroll start)
+  load_and_authorize_resource :course, class: "Course",
```

Comment:
```suggestion
  load_and_authorize_resource :course, class: Course.name,
```
Tương tự cho các class khác